### PR TITLE
Trigger workflow in remote (private) repo to build and upload private extensions

### PR DESCRIPTION
**Pull request by awestover**
Trigger workflow in remote (private) repo to build and upload private extensions

### CODE REVIEW

File: Dockerfile.compute-node [hunk 1]
```diff
@@ -579,6 +579,26 @@ RUN wget https://gitlab.com/dalibo/postgresql_anonymizer/-/archive/1.1.0/postgre
     sort -o /before.txt /before.txt && sort -o /after.txt /after.txt && \
     comm -13 /before.txt /after.txt | tar --directory=/usr/local/pgsql --zstd -cf /extensions/anon.tar.zst -T -
 
+#########################################################################################
```
Comment (vadim2404) [inline Dockerfile.compute-node:582]: + @bayandin 

Can we compile some extensions privately? I don't think that we can put them publicly for some of our customers.

Comment (bayandin) [reply Dockerfile.compute-node:582]: We can set up a CI workflow in a private repo and call it from the neon repo (the same we do it for e2e tests).

Comment (vadim2404) [reply Dockerfile.compute-node:582]: I like this idea, BTW. @awestover, can you make it real? :)

Comment (awestover) [reply Dockerfile.compute-node:582]: I'll try

**Comment by awestover** [comment]
@lubennikovaav gave some documentation for the enabling custom extensions part:

```
Add custom_extensions to user.settings
via admin UI or directly in the console database
!!! WARN !!! that this command overwrites current user settings. Use with caution.

update users set settings='{ "custom_extensions": ["my_custom_extname"] }' where email = 'anastasia@neon.tech';
if extension also needs to be added to shared_preload_libraries:
!!! WARN !!! that this command overwrites current user settings. Use with caution.

update users set settings='{ "pg_settings": {"shared_preload_libraries":"neon,anon,my_custom_extname"}, "custom_extensions": ["my_custom_extname"] }' where email = 'anastasia@neon.tech';
```

**Comment by awestover** [comment]
> @lubennikovaav gave some documentation for the enabling custom extensions part:
> 
> Add custom_extensions to `user.settings` via admin UI or directly in the console database !!! WARN !!! that this command overwrites current user settings. Use with caution. `update users set settings='{ "custom_extensions": ["my_custom_extname"] }' where email = 'user@email.com';`
> 
> if extension also needs to be added to `shared_preload_libraries`: !!! WARN !!! that this command overwrites current user settings. Use with caution. `update users set settings='{ "pg_settings": {"shared_preload_libraries":"neon,anon,my_custom_extname"}, "custom_extensions": ["my_custom_extname"] }' where email = 'user@email.com';`

actually, we will write some code to append to the json not just overwrite  it

**Comment by awestover** [comment]
Actually @vadim2404 I think it would make sense to move these docs to the repo where I'm working on private workflows for building extensions [neondatabase/build-custom-extensions](https://github.com/neondatabase/build-custom-extensions)
What do you think? Or is there another better place to put them?

**Comment by awestover** [comment]
@bayandin when you get a chance can you take a look at if I'm correctly calling the CI workflow from the neondatabase/build-custom-extension private repo?

File: Dockerfile.compute-node (hunk 1 above)
Comment (vadim2404) [reply Dockerfile.compute-node]: awesome!

Comment (awestover) [reply Dockerfile.compute-node]: oh actually right now it is flipped: the workflow gets skipped in remote repo when run from here, but gets run if run from there. need to unflip it

**Comment by awestover** [comment]
hopefully https://github.com/neondatabase/build-custom-extensions/pull/5 should fix this

**Comment by awestover** [comment]
This is expected to fail until https://github.com/neondatabase/build-custom-extensions/pull/5 is merged

**Comment by awestover** [comment]
currently this is broken because of yet another credential that I need in the build-private-extensions repo :upside_down_face:

**Comment by awestover** [comment]
should be fixed now

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -933,6 +933,46 @@ jobs:
...
+            "{
+              \"ref\": \"main\",
+              \"inputs\": {
+                \"ci_job_name\": \"build-and-upload-extensions\",
+                \"commit_hash\": \"$COMMIT_SHA\",
+                \"remote_repo\": \"${{ github.repository }}\",
+                \"compute_image_tag\": \"${{ needs.tag.outputs.build-tag }}\",
+                \"remote_branch_name\": \"${{ github.ref_name }}\"
+              }
+            }"
```
Comment (tristan957) [inline .github/workflows/build_and_test.yml:974]: you could save yourself some heartache here if you wrap the json in single quotes. Then you don't have to escape the double quotes.

Comment (tristan957) [reply .github/workflows/build_and_test.yml:974]: And then while you're here, might as well indent the continuation of the curl command.

Comment (tristan957) [reply .github/workflows/build_and_test.yml:974]: Also, have you seen https://github.com/actions/github-script? Alternative solution if you don't want to keep curl

Comment (awestover) [reply .github/workflows/build_and_test.yml:974]: thanks! I will use single quotes

Comment (awestover) [reply .github/workflows/build_and_test.yml:974]: although for some reason in the other places where we do this kind of stuff it's escaped double quotes

Comment (awestover) [reply .github/workflows/build_and_test.yml:974]: maybe for consistency I should either not change it or change in both places?
not sure if @bayandin has an opinion? (I think he wrote the other `curl` commands)

**Review by bayandin — APPROVED** [review]
Looks good, if it works, it works

File: .github/workflows/build_and_test.yml (hunk 2 above)
Comment (bayandin) [reply .github/workflows/build_and_test.yml:974]: We can also use `gh workflow run ...`.
Agree that this can be done in a separate PR.

**Comment by awestover** [comment]
I tested it from the other repo

**Pull request merged**